src/NuGet.Services.AzureSearch/Analysis/TruncateCustomTokenFilter.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Search.Models;
+
+namespace NuGet.Services.AzureSearch
+{
+    /// <summary>
+    /// Truncates tokens to 300 characters or less. This is necessary as Azure Search's
+    /// <see cref="TruncateTokenFilter"/> defaults to 10 characters.
```

Comment:
The REST API [documentation](https://docs.microsoft.com/en-us/azure/search/index-add-custom-analyzers#TokenFilters) states. Is there another place that states 10 as the default?

``
truncate | TruncateTokenFilter | Truncates the terms into a specific length.

Options 

length (type: int) - Default: 300, maximum: 300. Required.
``

Reply:
The `TruncateTokenFilter` passes `null` down to the API by default. The value of 10 isn't documented anywhere, it seems to be a bug (I've emailed the Azure Search folks about it)